agents/Group23/mcts.py

```diff
@@ -15,58 +16,52 @@ def __init__(self, colour: Colour, max_simulation_length: float = 2.5, custom_tr
         self.colour = colour  # Agent's colour
         self.max_simulation_length = max_simulation_length  # Length of a MCTS search in seconds
 
-    def _get_visit_count_distribution(self, node: TreeNode) -> list[list[int]]:
-        """Returns the visit count distribution for the children of the given node.
+    def run_simulation_with_process(self, root: TreeNode, colour: Colour,start_time: float) -> int:
+        """Runs a simulation with a new process."""
 
-        Args:
-            node (TreeNode): current node
+        iterations = 0
 
-        Returns:
-            list[list[int]]: visit count distribution
-        """
-        distribution_board = [[0 for _ in range(11)] for _ in range(11)]
-        self._count_visits_DFS(node, distribution_board)
-
-        # softmax normalization
-        total_visits = sum(sum(row) for row in distribution_board)
-        for i in range(11):
-            for j in range(11):
-                distribution_board[i][j] /= total_visits
-        return distribution_board
-    
-    def _count_visits_DFS(self, node: TreeNode, distribution_board: list[list[int]]):
-        """Counts the visits for the children of the given node.
+        while time.time() - start_time < self.max_simulation_length:
+            node = self._select(root)
+            result = self._simulate(node.board, colour)
+            self._backpropagate(node, result)
+            iterations += 1
 
-        Args:
-            node (TreeNode): current node
-            distribution_board (list[list[int]]): visit count distribution passed by reference
-        """
-        for child in node.children:
-            x, y = child.move.x, child.move.y
-            distribution_board[x][y] += child.visits
-            self._count_visits_DFS(child, distribution_board)
+        # delete second children
+        for child in root.children:
+            for grandchild in child.children:
+                del grandchild
 
+        return root, iterations
 
     def run(self, root: TreeNode):
         """Performs MCTS simulations from the root node."""
+        
+        _root = deepcopy(root)
+
         iterations = 0
         start_time = time.time()
-        while time.time() - start_time < self.max_simulation_length:
-            iterations += 1
-            node = self._select(root)
-            result = self._simulate(node.board, self.colour)
-            self._backpropagate(node, result)
+        number_of_workers = 8
```

Comment:
You did something clever before using the os module to get the number of cores if I remember correctly? On the off chance they decide to beef up the containers we run in, might be worth doing that here? That way we would exploit extra cores if they are given

Reply:
Sounds good :) I'll have a look on the other banch and pull that over